codewof/users/views.py

```diff
@@ -198,3 +219,378 @@ class UserAPIViewSet(viewsets.ReadOnlyModelViewSet):
+    else:
+        plaintext = "Hi,\n\n{} has invited you to join the Group '{}'. CodeWOF helps you maintain your " \
+                    "programming fitness with short daily programming exercises. With a free account you can save your " \
+                    "progress and track your programming fitness over time. Click the link below to make an account," \
+                    " using the email {}. You will see your invitation in the dashboard, where you can join the group. " \
+                    "If you already have a CodeWOF account, then add {} to your profile to make the invitation appear." \
+                    "\n\n{}\n\nThanks,\nThe Computer Science Education Research Group".format(inviter_name, group_name,
+                                                                                              email, email,
+                                                                                              reverse('account_signup'))
+    return plaintext
+
+
+def create_invitation_html(user_exists, invitee_name, inviter_name, group_name, email):
+    """
+    Builds the html for the invitation email, which is different depending if an account with the email exists or not.
+    :param user_exists: Whether a User object with this email exists.
+    :param invitee_name: The name of the invitee.
+    :param inviter_name: The name of the inviter.
+    :param group_name: The name of the group.
+    :param email: The email address.
+    :return:
+    """
+    email_template = get_template("users/group_invitation.html")
+    if user_exists:
+        message = "{} has invited you to join the Group '{}'. Click the link below to sign in. You will " \
+                  "see your invitation in the dashboard, where you can join the group.".format(inviter_name, group_name)
+        html = email_template.render({"user_exists": user_exists, "invitee_name": invitee_name, "message": message,
+                                      "url": reverse('account_login'), "button_text": "Sign In"})
+    else:
+        message = "{} has invited you to join the Group '{}'. CodeWOF helps you maintain your " \
+                  "programming fitness with short daily programming exercises. With a free account you can save your " \
+                  "progress and track your programming fitness over time. Click the link below to make an account," \
+                  " using the email {}. You will see your invitation in the dashboard, where you can join the group. " \
+                  "If you already have a CodeWOF account, then add {} to your profile to make the invitation appear." \
+            .format(inviter_name, group_name, email, email)
+        html = email_template.render({"user_exists": user_exists, "invitee_name": invitee_name, "message": message,
+                                      "url": reverse('account_signup'), "button_text": "Sign Up"})
+    return html
```

Comment:
These functions might be better in a `utils.py` file, since they aren't really a view in themselves.

Reply:
Done